### PR TITLE
Ported `cargo-check-each-crate-macos` to GitHub actions

### PR DESCRIPTION
**Pull request by Bullrich**
Resolves paritytech/ci_cd#1021

Ported the gitlab job to run on GitHub actions.

### CODE REVIEW

**Comment by Bullrich** [comment]
@alvicsam @lovelaced @mordamax this PR is ready for review.

I have tried both with `macos-latest` and `macos-latest-xlarge`

Both of them can complete the job. The first one takes more than 90 minutes, the second one takes around 45 minutes.

Which one should I assign as the runner for this job?

**Comment by lovelaced** [comment]
> @alvicsam @lovelaced @mordamax this PR is ready for review.
> 
> I have tried both with `macos-latest` and `macos-latest-xlarge`
> 
> Both of them can complete the job. The first one takes more than 90 minutes, the second one takes around 45 minutes.
> 
> Which one should I assign as the runner for this job?

What is the cost basis for each? (which one is cheaper to run given the time it takes :D?)

**Comment by Bullrich** [comment]
> > @alvicsam @lovelaced @mordamax this PR is ready for review.
> > I have tried both with `macos-latest` and `macos-latest-xlarge`
> > Both of them can complete the job. The first one takes more than 90 minutes, the second one takes around 45 minutes.
> > Which one should I assign as the runner for this job?
> 
> What is the cost basis for each? (which one is cheaper to run given the time it takes :D?)

From [the documentation](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates-for-standard-runners) it seems that the costs *per minute** are:
- `macos-latest`: $0.08
- `macos-latest-xlarge`: $0.16

So, seeing the time it takes, we could summarize the total price:
- `macos-latest`: Around $7.20 per run (90 minutes)
- `macos-latest-xlarge`: Around $7.20 per run (45 minutes)

As the `xlarge` cost double but takes half of the time they cost the same 🤣

**Review by EgorPopelyaev — APPROVED** [review]

File: .github/workflows/check-cargo-check-runtimes.yml [hunk 1]
```diff
@@ -120,6 +124,26 @@ jobs:
         with:
           root: cumulus/parachains/runtimes/testing
 
+  cargo-check-each-crate-macos:
```
Comment (bkchr) [inline .github/workflows/check-cargo-check-runtimes.yml:127]:
```suggestion
  cargo-check-all-crate-macos:
```

This is not using the script. Generally we only want to check that everything would compile on macos and using `cargo check --workspace` for that is fine.

**Comment by alvicsam** [comment]
I suggest closing this PR because [this one](https://github.com/paritytech/polkadot-sdk/pull/5690) was already merged

**Comment by Bullrich** [comment]
Closed in favour of #5690

**Pull request closed without merging**